logger/logger.go

```diff
@@ -25,195 +29,125 @@ var (
+	Warn(msg string, ctx *LogContext)
 }
 
-// TrailsLogger implements [Logger] using [*log.Logger].
+// TrailsLogger implements [Logger] using [golang.org/x/exp/slog.Logger].
 type TrailsLogger struct {
+	l    *slog.Logger
 	skip int
-	env  string
-	l    *log.Logger
-	ll   LogLevel
 }
 
-// New constructs a TrailsLogger.
-//
-// Logs are printed to [os.Stdout] by default, using the std lib log pkg.
-// The default environment is DEVELOPMENT.
-// The default log level is DEBUG.
-func New(opts ...LoggerOptFn) Logger {
-	logger := log.New(os.Stdout, "", log.LstdFlags)
-	l := &TrailsLogger{
-		env: "DEVELOPMENT",
-		l:   logger,
-		ll:  LogLevelInfo,
-	}
-	for _, opt := range opts {
-		opt(l)
-	}
-
-	if sentryDsn := os.Getenv("SENTRY_DSN"); sentryDsn != "" {
-		l.Info("SENTRY_DSN set, configuring SentryLogger", nil)
-		return NewSentryLogger(l, sentryDsn)
-	}
-
-	return l
-}
+// New constructs a Logger using [golang.org/x/exp/slog.Logger].
+func New(log *slog.Logger) Logger { return &TrailsLogger{l: log} }
 
-// AddSkip replaces the current number of frames to scroll back
-// when logging a message.
-//
-// Use Skip to get the current skip amount
-// when needing to add to it with AddSkip.
-func (l *TrailsLogger) AddSkip(i int) SkipLogger {
+func (l *TrailsLogger) AddSkip(i int) Logger {
 	newl := *l
 	newl.skip = i
 	return &newl
 }
 
-// Debug writes a debug log.
-func (l *TrailsLogger) Debug(msg string, ctx *LogContext) {
-	if l.ll > LogLevelDebug {
-		return
-	}
+func (l *TrailsLogger) Skip() int                         { return l.skip }
+func (l *TrailsLogger) Debug(msg string, ctx *LogContext) { l.log(slog.LevelDebug, msg, ctx) }
+func (l *TrailsLogger) Error(msg string, ctx *LogContext) { l.log(slog.LevelError, msg, ctx) }
+func (l *TrailsLogger) Info(msg string, ctx *LogContext)  { l.log(slog.LevelInfo, msg, ctx) }
+func (l *TrailsLogger) Warn(msg string, ctx *LogContext)  { l.log(slog.LevelWarn, msg, ctx) }
 
-	l.log(color.WhiteString, LogLevelDebug, msg, ctx)
-}
+// log executes printing the log message,
+// including any context if available.
+func (l *TrailsLogger) log(level slog.Level, msg string, ctx *LogContext) {
+	if ctx == nil {
+		ctx = new(LogContext)
+	}
 
-// Error writes an error log.
-func (l *TrailsLogger) Error(msg string, ctx *LogContext) {
-	if l.ll > LogLevelError {
-		return
+	pc := ctx.Caller
+	if pc == 0 {
+		// NOTE(dlk): skip the number of frames the TrailsLogger has
+		// and however many the TrailsLogger is configured with
+		pc, _, _, _ = runtime.Caller(knownFrames + l.Skip())
 	}
 
-	l.log(color.RedString, LogLevelError, msg, ctx)
+	rec := slog.NewRecord(time.Now(), level, msg, pc)
+	rec.AddAttrs(ctx.attrs()...)
+
+	l.l.Handler().Handle(context.TODO(), rec)
 }
 
-// Fatal writes a fatal log.
-func (l *TrailsLogger) Fatal(msg string, ctx *LogContext) {
-	if l.ll > LogLevelFatal {
-		return
+// ColorizeLevel adds color to the log level!
+func ColorizeLevel(groups []string, a slog.Attr) slog.Attr {
```

Comment:
To minimize the changes in our fork of [tint](https://github.com/xy-planning-network/tint), I opted to implement a `ReplaceAttr` function to add to [`slog.HandlerOptions`](https://pkg.go.dev/golang.org/x/exp/slog#HandlerOptions). While that fork includes changing the color of the log message, this function here makes it so that fork doesn't include changes to changing the log level color.